### PR TITLE
Upgrade Microsoft.NetCore.App to 2.0.3

### PR DESCRIPTION
**Pull request by KenitoInc**
### Issues

*This pull request fixes #xxx.*

### Description
PR builds in the WebAPI repo are failing
```
The nuget command failed with exit code(1) and error(NU1903: Warning As Error: Package 'Microsoft.NETCore.App' 2.0.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-7mfr-774f-w5r9
Errors in D:\a\1\s\test\UnitTest\Microsoft.AspNetCore.OData.Test\Microsoft.AspNetCore.OData.Test.csproj
    NU1903: Warning As Error: Package 'Microsoft.NETCore.App' 2.0.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-7mfr-774f-w5r9)
```

The vulnerability was fixed in v2.03 see https://github.com/advisories/GHSA-7mfr-774f-w5r9

I have upgraded to v2.0.3 and not a higher version to ensure all our code compiles as expected and tests pass.

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj [hunk 1]
```diff
@@ -80,6 +80,12 @@
       <DependentUpon>SRResources.resx</DependentUpon>
     </Compile>
   </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <PackageReference Include="Microsoft.Netcore.app">
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj:85]:
```suggestion
    <PackageReference Include="Microsoft.Netcore.App">
```

**Review by xuzhg — APPROVED** [review]

File: src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj (hunk 1 above)
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj:85]: Can we add this `PackageReference` where we have a similar condition (Line 18):
image

Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj:85]: Why do we need this package reference?

**Comment by KenitoInc** [comment]
Fixed by #2824

**Pull request closed without merging**